xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3599,7 +3598,8 @@ bool CVideoPlayer::OpenStream(CCurrentStream& current, int64_t demuxerId, int iS
       std::unique_ptr<CDVDDemuxVobsub> demux(new CDVDDemuxVobsub());
       if(!demux->Open(st.filename, source, st.filename2))
         return false;
-      m_pSubtitleDemuxer = demux.release();
+
+      m_pSubtitleDemuxer = std::move(demux);
```

Comment:
not sure about this but IMO this was wrong before. I don't think we want to keep the old demuxer in case of opening the new one fails.